### PR TITLE
Remove IViewsProvider and make view lookup to the feature provider

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
Partial fix to #4760

**Comment by pranavkm** [comment]
cc @javiercn

**Comment by pranavkm** [comment]
cc @rynowak

File: test/Microsoft.AspNetCore.Mvc.Razor.Test/Compilation/ViewsFeatureProviderTest.cs [hunk 1]
```diff
@@ -18,49 +20,89 @@ public void PopulateFeature_ReturnsEmptySequenceIfNoAssemblyPartHasViewAssembly(
...
+        {
+            private readonly Dictionary<AssemblyPart, Type> _containerLookup;
+
+            public TestableViewsFeatureProvider(Dictionary<AssemblyPart, Type> containerLookup)
+            {
+                _containerLookup = containerLookup;
+            }
+
+            protected override Type GetViewInfoContainerType(AssemblyPart assemblyPart)
+                => _containerLookup[assemblyPart];
```
Comment (javiercn) [inline test/Microsoft.AspNetCore.Mvc.Razor.Test/Compilation/ViewsFeatureProviderTest.cs]: nit: Do we put the arrow on a new line? It kinda looks weird :dizzy_face:

**Comment by javiercn** [comment]
LGTM :shipit: when @rynowak is happy

**Comment by davidfowl** [comment]
Nice! LGTM. Is there another issue to make views work generally with application parts?

**Comment by pranavkm** [comment]
@davidfowl  - https://github.com/aspnet/Mvc/issues/5144 one. I'll speak to @danroth27 and see if he has a particular design thought for it. We also have to figure out how we'd want to talk about embedded views versus precompiled views (and when we'd want to prefer one over the other).

**Comment by danroth27** [comment]
I wasn't envisioning any particular design for #5144. Isn't making views work generally with application parts goodness regardless of whether we do #5144?

**Comment by javiercn** [comment]
@pranavkm @davidfowl I would think that precompiled views supersede embedded views. Even if you had some sort of overriding mechanism in which you could drop in a file and pick that instead of the existing view, I don't see any advantage for anyone creating a library with views to choose embedding them vs precompiling them. (I would assume you would package both dlls (library + precompiled views) in the nuget package and get them into your app when you install the package.

**Comment by danroth27** [comment]
Using precompiled views with libraries has a couple of issues that I think still need to be thought through. One is that the project has to target netcoreapp1.0 in order to be able to precompile the views, which seems like a hack. Another is figuring out how pack will work when you have precompiled views. With embedded views I don't think you have these problems.

**Comment by danroth27** [comment]
That said, I agree that we don't want the support burden of two solutions that pretty much solve the same scenario. I lean towards closing #5144 and focusing on solving the remaining precompilation issues.

**Pull request closed without merging**